### PR TITLE
attempt to address paramiko connection errors

### LINKED ISSUE #2777: [BUG] - Flaky local integration tests failing with paramiko.ssh_exception.AuthenticationException
### Describe the bug

I've noticed that sporadically, our local integration tests might fail with the following error:

```
------------------------------ Captured log setup ------------------------------
2024-10-17 06:18:35      INFO 1944 transport: Connected (version 2.0, client AsyncSSH_2.13.2)
2024-10-17 06:18:35      INFO 1944 transport: Auth banner: b'Starting your server...'
2024-10-17 06:18:36      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:36      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:37      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:37      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:38      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:38      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:39      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:39      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:40      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:40      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:41      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:41      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:42      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:42      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:43      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:43      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:44      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:44      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:45      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:46      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:46      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:47      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:47      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:48      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:48      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:49      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:49      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:50      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:50      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:51      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:51      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:52      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:52      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:53      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:53      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:54      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:54      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:55      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:55      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:56      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:56      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:57      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:57      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:58      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:58      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:18:59      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:00      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:00      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:01      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:01      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:02      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:02      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:03      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:03      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:04      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:04      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:05      INFO 1944 transport: Auth banner: b'.'
2024-10-17 06:19:05      INFO 1944 transport: Auth banner: b'failed to start server on time!\n'
2024-10-17 06:19:05      INFO 1944 transport: Authentication (password) failed.
=========================== short test summary info ============================
ERROR tests/tests_deployment/test_jupyterhub_ssh.py::test_simple_jupyterhub_ssh - paramiko.ssh_exception.AuthenticationException: Authentication failed.
==================== 27 passed, 1 error in 73.40s (0:01:13) ====================
```

Usually a simple re-trigger of the GHA workflow will work. See for example:
* https://github.com/nebari-dev/nebari/actions/runs/11379267190/attempts/1?pr=2773 (**failed**)
* https://github.com/nebari-dev/nebari/actions/runs/11379267190/attempts/2?pr=2773 (**succeeded after re-trigger**)

### Expected behavior

Local integration tests should not fail because of an authentication error, specially if suggested changes in a PR don't involve that component.

### OS and architecture in which you are running Nebari

GHA

### How to Reproduce the problem?

This issue happens only sporadically and it's not consistently reproducible.

### Command output

_No response_

### Versions and dependencies used.

_No response_

### Compute environment

None

### Integrations

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by viniciusdc**
## Reference Issues or PRs
closes #2777 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: tests/tests_deployment/test_jupyterhub_ssh.py [hunk 1]
```diff
@@ -14,23 +14,52 @@
...
         "password": jupyterhub_access_token,
         "allow_agent": constants.PARAMIKO_SSH_ALLOW_AGENT,
         "look_for_keys": constants.PARAMIKO_SSH_LOOK_FOR_KEYS,
-        "auth_timeout": 5 * 60,
     }
 
-    ssh_client = paramiko.SSHClient()
-    ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    for timeout in retry_timeouts:
+        params["auth_timeout"] = timeout
```
Comment (viniciusdc) [inline tests/tests_deployment/test_jupyterhub_ssh.py:39]:
```suggestion
        params["auth_timeout"] = timeout
        params["banner_timeout"] = timeout
```

File: tests/tests_deployment/test_jupyterhub_ssh.py [hunk 2]
```diff
@@ -14,23 +14,53 @@
 TIMEOUT_SECS = 300
 
 
-@pytest.fixture(scope="function")
+@pytest.fixture(scope="session")
```
Comment (viniciusdc) [inline tests/tests_deployment/test_jupyterhub_ssh.py:17]:
```suggestion
@pytest.fixture(scope="module")
```

File: tests/tests_deployment/test_jupyterhub_ssh.py [hunk 3]
```diff
@@ -14,64 +15,80 @@
...
+) -> dict[str, str]:
+    command_delimiters = {}
+    for command in commands:
+        delimiter = uuid.uuid4().hex
+        command_delimiters[command] = delimiter
+        b = channel.send(f"echo {delimiter}start; {command}; echo {delimiter}end\n")
+        if b == 0:
+            print(f"Command '{command}' failed to send")
+    # Wait for the output to be ready before reading
+    time.sleep(wait_time)
```
Comment (dcmcand) [inline tests/tests_deployment/test_jupyterhub_ssh.py:71]: If output from a command is slow, we can add a wait.

File: tests/tests_deployment/test_jupyterhub_ssh.py [hunk 4]
```diff
@@ -80,52 +97,60 @@ def test_print_jupyterhub_ssh(paramiko_object):
...
+        "ls -la": ".bashrc",
+        "cat ~/.bashrc": "Managed by Nebari",
+        "cat ~/.profile": "Managed by Nebari",
+        "cat ~/.bash_logout": "Managed by Nebari",
+        # Ensure we don't copy over extra files from /etc/skel in init container
+        "ls -la ~/..202*": "No such file or directory",
+        "ls -la ~/..data": "No such file or directory",
+    }
+
+    outputs = run_command_list(commands_contain.keys(), channel, 30)
```
Comment (dcmcand) [inline tests/tests_deployment/test_jupyterhub_ssh.py:150]: Had to add a wait for the command output to show up.

**Review by marcelovilla — APPROVED** [review]

**Pull request merged**